root/static/scripts/edit/URLCleanup.js

```diff
@@ -389,6 +452,9 @@ const CLEANUPS = {
     type: LINK_TYPES.bookbrainz,
     clean: function (url) {
       return url.replace(/^(?:https?:\/\/)?(?:[^.]+\.)?bookbrainz\.org\/([^\/]*)\/([a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12})(?:[#\/?].*)?$/, "https://bookbrainz.org/$1/$2");
+    },
+    validate: function (url, id) {
+      return /^https:\/\/bookbrainz\.org\/[^\/]+\/[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$/.test(url);
```

Comment:
This could be tighten by adding the condition `_.includes(CLEANUPS.bookbrainz.type, id)` so to discard BookBrainz entities for other link types, see `amazon` and `youtube` for examples.